x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -23,16 +39,91 @@ export const SAVED_OBJECT_TYPES = {
   VISUALIZATION: 'visualization',
 };
 
+interface RawModuleConfig {
+  id: string;
+  title: string;
+  description: string;
+  type: string;
+  logoFile: string;
+  defaultIndexPattern: string;
+  query: any;
+  jobs: Array<{ file: string; id: string }>;
+  datafeeds: Array<{ file: string; job_id: string; id: string }>;
+  kibana: {
+    search: Array<{ file: string; id: string }>;
+    visualization: Array<{ file: string; id: string }>;
+    dashboard: Array<{ file: string; id: string }>;
+  };
+}
+
+interface MlJobStats {
+  jobs: MlJob[];
+}
+
+interface Config {
+  dirName: any;
+  json: RawModuleConfig;
+}
+
+interface Result {
+  id: any;
+  title: any;
+  query: any;
+  description: any;
```

Comment:
Wouldn't some of these types be `string`?

Reply:
Added types in https://github.com/elastic/kibana/pull/57206/commits/63ea75b825f7659b09b62f62ab420f33651637ed